app.js

```diff
@@ -165,46 +165,152 @@ $(document).ready(function () {
 
     $("#SubmitButton").on("click", function (event) {
         event.preventDefault();
-        if ($("#barOption").is(":checked")) {
-            // Function to take the user search input and display results on the map
-            function initializeSearch() {
-                currentLocation = new google.maps.LatLng(latitude, longitude);
-                map = new google.maps.Map(document.getElementById("map"), {
-                    center: currentLocation,
-                    zoom: 13
-                });
-                var searchInput = $("#SearchField")
-                    .val()
-                    .trim()
-                    .toLowerCase();
-
-                request = {
-                    location: currentLocation,
-                    radius: "400",
-                    query: searchInput + " cocktail"
-                };
-                service = new google.maps.places.PlacesService(map);
-                service.textSearch(request, callback);
+        if ($("#SearchField").val().length === 0) {
+            console.log("Modal should open");
+            // Get the modal
+            var modal = document.getElementById("myModal");
+
+            // Get the <span> element that closes the modal
+            var span = document.getElementsByClassName("close")[0];
+
+            // When the user clicks the button, open the modal 
+
+            modal.style.display = "block";
+
+
+            // When the user clicks on <span> (x), close the modal
+            span.onclick = function () {
+                modal.style.display = "none";
+            }
+
+            // When the user clicks anywhere outside of the modal, close it
+            window.onclick = function (event) {
+                if (event.target == modal) {
+                    modal.style.display = "none";
+                }
+            }
+        }
+        else {
+            if ($("#barOption").is(":checked")) {
```

Comment:
Remove the if statement  on 194 and make it an else if statement on 193

Reply:
The if statement on 194 was already there.  The else in 193 is supposed to encompass everything that you already have in the submit button on click event, triggered when there is input in the search bar.  Thus I dont think the else if is needed, right?  I think that might resolve the issue directly below